pkg/platform/local/platform.go

```diff
@@ -1069,24 +1092,40 @@ func (p *Platform) deletePreviousContainers(createFunctionOptions *platform.Crea
 		return 0, errors.Wrap(err, "Failed to get function containers")
 	}
 
-	// if the function exists, delete it
-	if len(containers) > 0 {
-		createFunctionOptions.Logger.InfoWith("Function already exists, deleting function containers",
+	if createFunctionOptions.FunctionConfig.Spec.Disable {
+		createFunctionOptions.Logger.InfoWith("Disabling function",
 			"functionName", createFunctionOptions.FunctionConfig.Meta.Name)
 
-		// iterate over containers and delete
+		// if function is disabled, stop all containers
 		for _, container := range containers {
-			createFunctionOptions.Logger.DebugWith("Deleting function container",
+			createFunctionOptions.Logger.DebugWith("Stop function container",
 				"functionName", createFunctionOptions.FunctionConfig.Meta.Name,
-				"containerName", container.Name)
-			previousHTTPPort, err = p.getContainerHTTPTriggerPort(&container)
-			if err != nil {
-				return 0, errors.Wrap(err, "Failed to get a container's HTTP-trigger port")
+				"containerID", container.ID)
+			if err := p.dockerClient.StopContainer(container.ID); err != nil {
+				return 0, errors.Wrap(err, "Failed to stop a container")
 			}
+		}
+		return 0, nil
+	}
```

Comment:
What does it mean that the function is disabled?
And if we're only stopping and not deleting a disabled function, does it mean that its container is not using the HTTP Port?

Reply:
disabling a function means stopping it (weird naming but true to facts)
when function is stopped/paused, the docker container is stopped as well, which means, no http requests can be served